src/parsers/graph/model-graph.cpp

```diff
@@ -0,0 +1,277 @@
+//
+// Copyright (c) 2025 INRIA
+//
+
+#include "pinocchio/parsers/graph/model-graph.hpp"
+#include "pinocchio/parsers/graph/model-graph-algo.hpp"
+#include "pinocchio/parsers/graph/graph-visitor.hpp"
+#include "pinocchio/parsers/graph/joints.hpp"
+
+namespace pinocchio
+{
+  namespace graph
+  {
+    namespace
+    {
+      /// \return true if \p joint_name is already used in \p graph
+      bool isJointNameExists(const ModelGraph::Graph & graph, const std::string & joint_name)
+      {
+        for (auto e_it = boost::edges(graph); e_it.first != e_it.second; ++e_it.first)
+        {
+          if (graph[*e_it.first].name == joint_name)
+          {
+            return true;
+          }
+        }
+        return false;
+      }
+    } // namespace
+
+    template<int Nq, int Nv>
+    void JointLimits::setDimensions()
+    {
+      const double infty = std::numeric_limits<double>::infinity();
+
+      maxEffort = Eigen::VectorXd::Constant(Nv, infty);
+      maxVel = Eigen::VectorXd::Constant(Nv, infty);
+      maxConfig = Eigen::VectorXd::Constant(Nq, infty);
+      minConfig = Eigen::VectorXd::Constant(Nq, -infty);
+      friction = Eigen::VectorXd::Constant(Nv, 0.);
+      damping = Eigen::VectorXd::Constant(Nv, 0.);
+      armature = Eigen::VectorXd::Constant(Nv, 0);
+    }
+
+    void JointLimits::append(const JointLimits & range, const int nq, const int nv)
+    {
+      assert(range.maxEffort.size() == nv);
+      assert(range.minConfig.size() == nq);
+
+      maxEffort.conservativeResize(maxEffort.size() + nv);
+      maxEffort.tail(nv) = range.maxEffort;
+      maxVel.conservativeResize(maxVel.size() + nv);
+      maxVel.tail(nv) = range.maxVel;
+
+      minConfig.conservativeResize(minConfig.size() + nq);
+      minConfig.tail(nq) = range.minConfig;
+      maxConfig.conservativeResize(maxConfig.size() + nq);
+      maxConfig.tail(nq) = range.maxConfig;
+
+      damping.conservativeResize(damping.size() + nv);
+      damping.tail(nv) = range.damping;
+      friction.conservativeResize(friction.size() + nv);
+      friction.tail(nv) = range.friction;
+
+      armature.conservativeResize(armature.size() + nv);
+      armature.tail(nv) = range.armature;
+    }
+
+    EdgeBuilder & EdgeBuilder::withJointType(const JointVariant & jtype)
+    {
+      param.joint = jtype;
+      param.jlimit = boost::apply_visitor(internal::MakeJointLimitsDefaultVisitor(), jtype);
```

Comment:
You should do that in build:
- If you call `withMinConfig` and then `withJointType` you will erase the withMinConfig call.